gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -30,4 +39,20 @@ public static Configuration newConfiguration() {
 
     return conf;
   }
+
+  public static List<FileStatus> listStatusRecursive(FileSystem fileSystem, Path path) throws FileNotFoundException, IOException {
+    List<FileStatus> results = Lists.newArrayList();
+    walk(results, fileSystem, path);
+    return results;
+  }
+
+  private static void walk(List<FileStatus> results, FileSystem fileSystem, Path path) throws FileNotFoundException, IOException {
+    for (FileStatus status : fileSystem.listStatus(path)) {
```

Comment:
listStatus shouldn't return null. It should throw FileNotFoundException if path is not found.